Mathlib/LinearAlgebra/QuadraticForm/Basic.lean

```diff
@@ -315,172 +320,187 @@ theorem polar_sub_right (x y y' : M) : polar Q x (y - y') = polar Q x y - polar
 #align quadratic_form.polar_sub_right QuadraticForm.polar_sub_right
 
 @[simp]
-theorem polar_self (x : M) : polar Q x x = 2 * Q x := by
-  rw [polar, map_add_self, sub_sub, sub_eq_iff_eq_add, ← two_mul, ← two_mul, ← mul_assoc]
+theorem polar_self (x : M) : polar Q x x = 2 • Q x := by
+  rw [polar, map_add_self, sub_sub, sub_eq_iff_eq_add, ← two_smul ℕ, ← two_smul ℕ, ← mul_smul]
   norm_num
 #align quadratic_form.polar_self QuadraticForm.polar_self
 
 /-- `QuadraticForm.polar` as a bilinear form -/
 @[simps]
-def polarBilin : BilinForm R M where
-  bilin := polar Q
-  bilin_add_left := polar_add_left Q
-  bilin_smul_left := polar_smul_left Q
-  bilin_add_right x y z := by simp_rw [polar_comm _ x, polar_add_left Q]
-  bilin_smul_right r x y := by simp_rw [polar_comm _ x, polar_smul_left Q]
+def polarBilin : M →ₗ[R] M →ₗ[R] N where
+  toFun (a : M) := ⟨{
+    toFun := polar Q a
+    map_add' := polar_add_right Q a
+  }, by simp⟩
+  map_add' _ _ := by
+    ext
+    simp
+  map_smul' _ _ := by
+    ext
+    simp
```

Comment:
I think we're going to want separate definitions (`polarBilin` and `polarBilinForm`) in the short term.

Note that you can use `LinearMap.mk\2` here to golf this.

Reply:
I've changed it to use `LinearMap.mk₂`. The def is now
```
def polarBilin : BilinMap R M N
```
where
```
variable (R M Nₗ) in
/-- A shorthand for the type of `R`-bilinear `N`-valued maps on `M`. -/
protected abbrev BilinMap : Type _ := M →ₗ[R] M →ₗ[R] Nₗ

variable (R M) in
/-- For convenience, a shorthand for the type of bilinear forms from `M` to `R`.

This should eventually replace `_root_.BilinForm`. -/
protected abbrev BilinForm : Type _ := LinearMap.BilinMap R M R
```
@eric-wieser following #11278 do we still need a separate `polarBilinForm`?

Reply:
I think probably not.